### PR TITLE
Removed unused imports and variables and added future imports

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
Hi,

Removed unused imports and variables and added future imports

Marcus

### CODE REVIEW

File: mxcube3/core/qutils.py [hunk 1]
```diff
@@ -1330,7 +1332,7 @@ def add_data_collection(node_id, task):
     dc_model, dc_entry = _create_dc(task)
     set_dc_params(dc_model, dc_entry, task, sample_model)
 
-    pt = dc_model.acquisitions[0].path_template
+    dc_model.acquisitions[0].path_template
```
Comment (meguiraun) [inline mxcube3/core/qutils.py]: is this line needed at all?
same for the next ones

File: mxcube3/core/utils.py [hunk 2]
```diff
@@ -103,7 +106,7 @@ def get_light_limits():
 
 
 def get_movable_state_and_position(item_name):
-    ret = dict()
+    pass
```
Comment (meguiraun) [inline mxcube3/core/utils.py]: why this?

File: mxcube3/routes/mockups.py [hunk 3]
```diff
@@ -29,12 +29,11 @@ def mockup_newres(newres):
 def create_diff_plan(sid):
     """Juts for creating a diff plan as if it were created by edna and so on.
     """
-    from mock import Mock
 
     acq_parameters = blcontrol.beamline.get_default_acquisition_parameters()
     ftype = blcontrol.beamline.detector_hwobj.getProperty("file_suffix")
     ftype = ftype if ftype else ".?"
-    n = int(blcontrol.session["file_info"].getProperty("precision", 4))
+    int(blcontrol.session["file_info"].getProperty("precision", 4))
```
Comment (meguiraun) [inline mxcube3/routes/mockups.py]: should go out?

File: mxcube3/routes/signals.py [hunk 4]
```diff
@@ -163,7 +167,7 @@ def loaded_sample_changed(sample):
 
     try:
         # recreate the dict with the sample info
-        q = queue_to_dict()
+        queue_to_dict()
```
Comment (meguiraun) [inline mxcube3/routes/signals.py]: out

File: mxcube3/routes/signals.py [hunk 5]
```diff
@@ -384,9 +388,9 @@ def collect_oscillation_failed(
 
     if not qutils.is_interleaved(node["node"]):
         try:
-            limsres = blcontrol.rest_lims.get_dc(lims_id)
+            blcontrol.rest_lims.get_dc(lims_id)
         except BaseException:
-            limsres = {}
+            pass
```
Comment (meguiraun) [inline mxcube3/routes/signals.py]: this try bloc is not needed

**Comment by meguiraun** [comment]
thanks for this! I only noticed small thins, basically lines no longer needed (I guess you removed the unused var)

File: mxcube3/routes/signals.py (hunk 5 above)
Comment (marcus-oscarsson) [reply mxcube3/routes/signals.py]: The call to get_dc can still go wrong so the try block would still be needed, unless you see a scenario when this never will go wrong ?

**Comment by marcus-oscarsson** [comment]
Hi Mikel, 

Exactly I removed a few unused variables. As you noticed  I missed to remove some of the calls without side effects. However that's taken care of now.

Marcus

**Comment by meguiraun** [comment]
thanks!

File: mxcube3/routes/signals.py (hunk 5 above)
Comment (meguiraun) [reply mxcube3/routes/signals.py]: sorry, I missed this comment. I meant the actual call is not needed in this method

**Pull request merged**